lib/cloudregister/registerutils.py

```diff
@@ -531,31 +531,35 @@ def setup_registry(registry_fqdn, username, password):
 
 
 # ----------------------------------------------------------------------------
-def get_registry_credentials():
+def get_registry_credentials(set_new):
     """Read the registry credentials file
        and return its content or an empty dict."""
     config_json = {}
-    failed = False
-    if os.path.exists(REGISTRY_CREDENTIALS_PATH):
-        try:
-            with open(REGISTRY_CREDENTIALS_PATH, 'r') as cred_json:
-                config_json = json.load(cred_json)
-        except json.decoder.JSONDecodeError:
-            logging.info(
-                'Unable to parse existing %s, preserving file as %s.bak, '
-                'writing new credentials' % (
-                    REGISTRY_CREDENTIALS_PATH, REGISTRY_CREDENTIALS_PATH
-                )
-            )
-            # we want to remove the possible
-            # auth token present in the file
-            # before preserving it
-            mv_file_cmd = 'mv -Z {} {}.bak'.format(
-                REGISTRY_CREDENTIALS_PATH, REGISTRY_CREDENTIALS_PATH
-            ).split()
-            failed = exec_subprocess(mv_file_cmd)
-            message = 'File not preserved.' if failed else 'File preserved.'
-            logging.info(message)
+    failed = None
+    if not os.path.exists(REGISTRY_CREDENTIALS_PATH):
+        logging.info('Credentials file does not exist. Nothing to do')
+        return config_json, failed
+
+    try:
+        with open(REGISTRY_CREDENTIALS_PATH, 'r') as cred_json:
+            config_json = json.load(cred_json)
+    except json.decoder.JSONDecodeError:
+        message = (
+            'Unable to parse existing %s, preserving file as %s.bak'
+            % (REGISTRY_CREDENTIALS_PATH, REGISTRY_CREDENTIALS_PATH)
+        )
+        if set_new:
+            message += ', writing new credentials'
+        logging.info(message)
+        # we want to remove the possible
+        # auth token present in the file
+        # before preserving it
+        mv_file_cmd = 'mv -Z {} {}.bak'.format(
+            REGISTRY_CREDENTIALS_PATH, REGISTRY_CREDENTIALS_PATH
+        ).split()
+        failed = exec_subprocess(mv_file_cmd)
+        message = 'File not preserved' if failed else 'File preserved'
+        logging.info(message)
```

Comment:
We need our own exec_subprocess to move a file ? How about ```shutil.move```. It's also unclear to me why the selinux fattr needs to be reset to the default. It actually feels wrong to do that if there is a selinux context set on the file. If this is wanted the comment should rather state the reason for the use of the mv options instead of saying in text what the next line does :)

Reply:
> We need our own exec_subprocess to move a file ? How about shutil.move

the suggested method to use was `os.system` and call a cmd, happy to use `shutil.move`, 
if we are calling a cmd, yes  it is my opinion to use the same method for that `exec_subprocess` in this case

Reply:
`os.system` was an example, anyway I agree we already have a subprocess wrapper we should use it.

With respect to the SELinux context, I am not sure how that works when we create the file or if a user created the file. We get here when we cannot read the existing config file, for whatever reason and since we create a backup maybe we do not care for the SELinux context. That said, this was part of the previous PR and is an artifact in this diff because this branch has not yet been rebased. An artifact of chaining branches and then not waiting until the start of the chain is merged and rebasing before sending a new PR. Which makes review more difficult.

Reply:
I applied the changes from master on this branch yesterday

> Which makes review more difficult

Yes, sorry about that, it's a side effect of the time constraint we have for this feature
Hopefully, it is not that much more difficult

Reply:
> anyway I agree we already have a subprocess wrapper we should use it.

@rjschwei I understand you prefer `exec_subprocess` than using `os.rename` for this case, correct ?
or maybe not prefer as a strong word but maybe more "let's make fewer changes"

Reply:
I'm fine either way. If you still need the SELinux context related move, let's stick to how it is right now